hazelcast/src/main/java/com/hazelcast/cache/impl/operation/CachePutAllBackupOperation.java

```diff
@@ -35,14 +40,14 @@
  *
  * @see com.hazelcast.cache.impl.operation.CacheLoadAllOperation
  */
-public class CachePutAllBackupOperation extends CacheOperation implements BackupOperation {
+public class CachePutAllBackupOperation extends CacheOperation implements BackupOperation, Versioned {
 
-    private Map<Data, CacheRecord> cacheRecords;
+    private Map<Data, WanWrappedCacheRecord> cacheRecords;
```

Comment:
What about declaring `cacheRecords` as `Map<Data, ? extends CacheRecord>`?
This way we wouldn't impose record-wrapping to all operations (eg loadAll etc) and their serialized form. The disadvantage is there will be an additional complexity (`instanceof` check, conditionally run/serialize the `isWanReplicated` logic), so maybe it's not a great trade-off. wdyt?

Reply:
I'd actually like to do what I originally thought about doing - converting the Cache backup operations to use a list of pairs, like Map operations do; that way I don't need to have any wrapper at all, and everything can be dealt with in the same manner as Map operations are.

For PoC purposes and as it wasn't difficult to do, I made the changes in this commit: https://github.com/hazelcast/hazelcast/pull/24928/commits/eccd294359740ac33acd63fb8fee65ea0806d213